lib/solr_wrapper/instance.rb

```diff
@@ -222,17 +226,23 @@ def with_collection(options = {})
     end
 
     ##
-    # Clean up any files solr_wrapper may have downloaded
-    def clean!
+    # Clean up any files solr_wrapper may have downloaded, except the solr-{ver}.zip file
+    def clean
```

Comment:
Here is our previous discussion:

https://github.com/cbeer/solr_wrapper/pull/104#discussion_r115878079

```
@cbeer
cbeer 12 days ago Owner
Instead of removing this line whole-sale, let's put it behind a named parameter flag (say, #clean! except_download_dir: true, or something better). There's a real use-case for "get rid of everything solr_wrapper has done", and I want to preserve that ability.
       @darrenleeweber
darrenleeweber 12 days ago
clean vs. clean! maybe? where the latter calls this clean and then also cleans up the downloads (all of them).
       @cbeer
cbeer 12 days ago Owner
A ! suffix usually indicates dangerousness or mutating state, neither of which seem to be true here. I'd rather we make it an opt-in/-out flag.
```

I see you chose to use `clean` and `clean!` instead, which surprises me.